src/main.cpp

```diff
@@ -4503,9 +4503,12 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                 }
                 if (!pushed && inv.type == MSG_TX) {
                     CTransaction tx;
-                    if (mempool.lookup(inv.hash, tx)) {
-                        pfrom->PushMessage(NetMsgType::TX, tx);
-                        pushed = true;
+                    int64_t txtime;
+                    if (mempool.lookup(inv.hash, tx, txtime)) {
+                        if (txtime <= pfrom->timeLastMempool) {
```

Comment:
Indeed, current layout is an artifact of my local copy having an else for logging.